src/jaxsim/utils/wrappers.py

```diff
@@ -76,3 +109,37 @@ def __eq__(self, other: HashedNumpyArray) -> bool:
             return np.array_equal(self.array, other.array)
 
         return hash(self) == hash(other)
+
+    @staticmethod
+    def hash_of_array(
+        array: jax.Array | npt.NDArray, precision: float | None = 1e-9
+    ) -> int:
+        """
+        Calculate the hash of a NumPy array.
+
+        Args:
+            array: The array to hash.
+            precision: Optionally limit the precision over which the hash is computed.
```

Comment:
I'm not that familiar with this logic. Why is this necessary?

Reply:
The logic before this PR had no precision. For low-level reasons I didn't investigate, hashing numpy arrays was only persistent for a couple of seconds, then it was changing to another value. This behaviour puzzled me quite a lot.

The new logic is done in such a way that the hash of numpy arrays is always stable up to the given precision (both for the integer and the decimal parts). And, furthermore, this default should be also compatible in 32-bit precision without overflow.

Reply:
In general, hashing numpy arrays is always a bad idea. And this is the reason why `np.all_close` is used for arrays of floating point numbers, that under the hood uses absolute/relative tolerances.